zbus_macros/src/proxy.rs

```diff
@@ -619,12 +618,23 @@ fn gen_proxy_method_call<M: AttrParse + Into<MethodAttrs>>(
         let is_input_type = inputs.iter().any(|arg| {
             // FIXME: We want to only require `Serialize` from input types and `DeserializeOwned`
             // from output types but since we don't have type introspection, we employ this
-            // workaround of regex matching on string reprepresention of the the types to figure out
+            // workaround of matching on the string representation of the the types to figure out
             // which generic types are input types.
             if let FnArg::Typed(pat) = arg {
-                let pattern = format!("& *{}", param.to_token_stream());
-                let regex = Regex::new(&pattern).unwrap();
-                regex.is_match(&pat.ty.to_token_stream().to_string())
+                let pat = pat.ty.to_token_stream().to_string();
+                let param = param.to_token_stream().to_string();
+
+                let mut found = false;
+                for part in pat.split('&').skip(1) {
+                    let part = part.trim_start_matches(' ');
+
+                    if part == param {
+                        found = true;
+                        break;
+                    }
+                }
```

Comment:
Actually it's strange to split by `&` when it will be the first char. I think this could be simpler. Maybe something like:

```suggestion
                if let Some('&') = pat.chars().next() {
                    let ty_name = pat[1..].trim_start();
                    
                    ty_name == param
                } else {
                    false
                }
```

Reply:
That's what I initially implemented, but I was mimicking what the code does and I wasn't sure of all the possible uses or how where/when syn inserts random whitespace (since there's only two or so tests for this). I can swap back to that if you want.

Reply:
Given that my suggestion is doing exactly what the regex-based code was doing and is simpler IMO, I'd prefer that, yes.